### ISSUE TITLE
set dialog width?

### ISSUE THREAD

**Issue by zsj1029** [biig-io/ngx-smart-modal#103]
version:6.0

how to modify dialog width?
i set 
<code>
 .nsm-dialog {
   &.nsm-dialog-open
 {     width: 75% ;  
 }
}
</code>

in sub-Component's scss file

but its not useful

i only could set it in source file "ngx-smart-modal/ngx-smart-modal.scss"
<code>@import "~ngx-smart-modal/ngx-smart-modal";</code>
it's useful

but it will be Global style,i need part style

sorry my English is not good

thx for this goods

**Comment by khylias** [comment]
Hi @zsj1029,

By default, Angular Component styles are encapsulated in component scope, then they don't affect the rest of the application, [check out the documentation ](https://angular.io/guide/component-styles#view-encapsulation). 😃 
In your case, you can set the `encapsulation` of subComponent at `ViewEncapsulation.None`. But keep warning, this expose all subComponent's styles at the application.

I recommande you to use `customClass` property of Ngx-Smart-Modal and set it on global styles.

**Comment by maximelafarie** [comment]
I agree with @khylias,
I suggest you to import `ngx-smart-modal` style (`@import "~ngx-smart-modal/ngx-smart-modal";`) in the main `.scss` file of your projet (default `styles.scss`).

It will expose all the library style across your app. So you can override some variables according to the [documentation](https://github.com/biig-io/ngx-smart-modal#style--customization) but keep in mind it will override the library style variables **globally** (for all modal instances). 

To sum up, it's only interesting to override variables to change all your modals style. But it won't be pertinent if it customizes only one or two modals.

As @khylias said, you can create some custom classes inside your app `styles.scss` file (or any `scss` file imported into it) that overrides library styles. Then, set your custom class inside modal `customClass` option.

This issue also could be related to issue #101 (comment about setting modal width as a SCSS variable)

**Comment by zsj1029** [comment]
@khylias 
thanks a lot  , I found it.
this is my situation。
1. I use Ngx-Smart-Modal in my owner component 
2. I set customClass property  on Ngx-Smart-Modal html tag.

<code>
<ngx-smart-modal customClass="msg-modal" #myModal identifier="myModal">
</ngx-smart-modal>
</code>

3. I set style css in my component for customClass, and try to override "ngx-smart-modal" width, but its not work
4. I also set customClass width css in  global styles( styles.scss ), not work.

**Comment by khylias** [comment]
Have you overwrite the `max-width` property too ? By default, in `ngx-smart-modal.scss` it set at `520px`

**Comment by zsj1029** [comment]
@khylias @maximelafarie 

> To sum up, it's only interesting to override variables to change all your modals style. But it won't be 
> pertinent if it customizes only one or two modals.

yeah, thx a lot.

**Comment by zsj1029** [comment]
@maximelafarie @khylias 
1. ```<ngx-smart-modal customClass="msg-modal" #myModal identifier="myModal">```
2. <code>@import "~ngx-smart-modal/ngx-smart-modal";
@import url("./assets/css/ngx-cust-styles.scss");</code> 
in my root styles.scss
3. <code>.msg-modal {
  width: 75%;
  max-width: 520px;
  border-radius: 10px;
  padding: 12px 0 0 0;
}</code> in ngx-cust-styles.scss, my self style file

then not work, sadness story

**Comment by khylias** [comment]
In relation to which element do you base the 75% of your modal?
Let's say you have a browser window open at 1440px * 900px.
You want the modal to make 75% of your window, that's equal to 1080px. 
As the modal has `max-width` property at `520px`, your modal will not exceed 520px wide. You must set the value to none to grow up your modal at 1080px.

**Comment by zsj1029** [comment]
May i ask any other questions?
about url params


When i view my app by url like
http://127.0.0.1/?token=xxxxxxxxx


I want to get token in the app.component.ts
Please give me a train of thought to solve this problem








On 4/20/2018 17:16，Kraus Vincent<notifications@github.com> wrote：

In relation to which element do you base the 75% of your modal?
Let's say you have a browser window open at 1440px * 900px.
You want the modal to make 75% of your window, that's equal to 1080px.
As the modal has max-width property at 520px, your modal will not exceed 520px wide. You must set the value to none to grow up your modal at 1080px.

**Comment by maximelafarie** [comment]
@zsj1029 Please open a new issue if you question concerns `ngx-smart-modal`.

If it doesn't, please ask your question on [stackoverflow](https://stackoverflow.com/) or equivalent. 😉
